File: Rust/fib.rs
// rustc -O fib.rs

use std::time::Instant;

fn fib_iterative(n: u32) -> u32 {
    let mut a = 0;
    let mut b = 1;
    for _ in 0..n {
        let c = a + b;
        a = b;
        b = c;
    }
    a
}

fn fib_recursive(n: u32) -> u32 {
    if n <= 1 {
        return n;
    }
    fib_recursive(n - 1) + fib_recursive(n - 2)
}

fn main() {
    let now = Instant::now(); // start timer
    println!("{}", fib_recursive(42));
    let elapsed = now.elapsed(); // check time elapsed
    let sec = (elapsed.as_secs() as f64) + (elapsed.subsec_nanos() as f64 / 1_000_000_000.0);
    println!("Seconds Recursive took: {:.3}", sec);

    let now = Instant::now(); // start timer
    println!("{}", fib_iterative(42));
    let elapsed = now.elapsed(); // check time elapsed
    let sec = (elapsed.as_secs() as f64) + (elapsed.subsec_nanos() as f64 / 1_000_000_000.0);
    println!("Seconds Iterative took: {:.3}", sec);
}

File: Go/fib.go
// go build fib.go

package main

import (
	"fmt"
	"time"
)

func fib_recursive(n uint64) uint64 {
	if n <= 1 {
		return n
	}
	return fib_recursive(n-1) + fib_recursive(n-2)
}

func fib_iterative(n int) int {
	a := 0
	b := 1
	for i := 1; i <= n; i++ {
		c := a + b
		a, b = b, c
	}
	return a
}

func main() {
	start := time.Now() // Start timer
	fmt.Println(fib_recursive(42))
	duration := time.Since(start) // End timer & duration
	elapsed := float64(duration) / float64(time.Millisecond)
	elapsed = elapsed / 1000.0
	fmt.Printf("Recursive took secs: %.3f\n", elapsed)

	start = time.Now() // Start timer
	fmt.Println(fib_iterative(42))
	duration = time.Since(start) // End timer & duration
	elapsed = float64(duration) / float64(time.Millisecond)
	elapsed = elapsed / 1000.0
	fmt.Printf("Iterative took secs: %.3f\n", elapsed)

}
